assets/scripts.js

```diff
@@ -20,4 +19,79 @@ $(function () {
       listUrls.addClass('is--visible');
     }
   });
-})
+
+  // Adding copy buttons to all pre tags parentNode and add class code-container
+  var preBlocks = document.getElementsByTagName('pre');
+  for (var i=0;i<preBlocks.length;i++) {
+
+    // console.log(preBlocks[i].parentNode);
+    var div = document.createElement('div'); // Creating a new node
+
+    // Adding properties to the div
+    div.setAttribute('class','copy-code-container');
+    div.innerHTML = '<button class="btn btn-light code-copy-button" data-toggle="tooltip" data-placement="top" title="Copy to clipboard" onclick=copyThisCode(this)>Copy</button>';
+
+    // Prepending and adding classes to the parentNode
+    preBlocks[i].parentNode.prepend(div); // prepending the new node
+    preBlocks[i].parentNode.classList.add('code-container');
+    console.log("Successfully added copy buttons to the codes");
+  }
+
+  // Use this after adding above code triggered when your mouse enters the .code-container block
+  // <!-- Script for showing and hiding code button -->
+  $('.code-container').hover(function() {
+
+    // console.log(this);
+    // console.log(this.parentNode); // This is the outer div which containes the pre
+
+    if(this.classList.contains('is-hovered')){
+
+      // console.log("Mouse leaved outside ");
+      this.classList.remove('is-hovered');
+
+    } else {
+
+      // console.log("Mouse entered inside ");
+      this.classList.add('is-hovered');
+
+    }
+  });
+
+  // For toggle, added this line after all data-toggle needed are done
+  $('[data-toggle="tooltip"]').tooltip();
+
+});
+
+// Pass 'this' to this function of the button and this will copy the code
+function copyThisCode(btnClicked) {
+
+  // Checking some values as correct or not
+  // console.log(btnClicked);
+  var codeContainer = btnClicked.parentNode.parentNode;
+
+  // Get the code which need to be copied
+  var codeToCopy = codeContainer.getElementsByTagName('pre')[0].textContent;
+  // console.log(codeToCopy);
+
+  // Create a temporary textarea to select and copy to clipboard
+  var tempInput = document.createElement("textarea");
```

Comment:
There is no need to create a `tempInput` HTML element, just simply use `navigator.clipboard.writeText(codeToCopy);` instead of `document.execCommand("copy");`, and the code will be copied.

Reply:
Regarding this one, I have confusion.. discussing on slack.

Reply:
Done